packages/definitions-parser/src/lib/module-info.ts

```diff
@@ -253,10 +253,7 @@ function findReferencedFiles(src: ts.SourceFile, packageName: string, subDirecto
       full = full.slice(packageName.length + 4);
     } else if (baseDirectory && full.startsWith("../" + baseDirectory + "/")) {
       full = full.slice(baseDirectory.length + 4);
-    } else if (
-      full.startsWith("..") &&
-      (baseDirectory === "" || path.normalize(joinPaths(baseDirectory, full)).startsWith(".."))
-    ) {
+    } else if (full.startsWith(baseDirectory ? "../.." : "..")) {
```

Comment:
Is this equivalent? When `baseDirectory !== ""`, the original checks `joinPaths(baseDirectory, full)`, but the new version checks `full` with a longer initial string.

This could be obviously correct if `full` were produced from `baseDirectory`, but it's produced from `subDirectory`. Is there some equivalence of those two that I'm missing?

Reply:
Assuming `full` is already normalized (it is), then the following are equivalent
- `path.normalize(joinPaths("segment", full)).startsWith("..")`
- `full.startsWith("../..")`

If ``path.normalize(`segment/${full}`)`` starts with `".."` then `full` must have started with `"../.."`, to have soaked up `"segment"` and still start with `".."`, and if `full` starts with `"../.."` then ``path.normalize(`segment/${full}`)`` will still start with `".."`.

So, as long as `baseDirectory` isn't `"."` or `".."` (it isn't, it's `"ts2.0"` or the like), then this change is equivalent.